src/screens/Internal/useInternalController.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {useNavigation, useRoute} from '@react-navigation/native';
+import {format} from 'date-fns';
+
+import {ArticleJSON} from '~/@types/dtos/everythingDTO';
+import {StyledInternal} from './styles';
+
+// TODO - Try to send styles from useInternalController params
+import {styles} from './index';
+
+import HeaderButton from '~/components/HeaderButton';
+import Row from '~/components/Row';
+import RowBetween from '~/components/RowBetween';
+import Separator from '~/components/Separator';
+import Text from '~/components/Text';
+
+const useInternalController = () => {
+  const {goBack} = useNavigation();
+  const navigationParams = useRoute().params;
+  // TODO - Organize the screen route params an their params
+  const news: ArticleJSON = navigationParams!.item;
+  const {IconContainer} = StyledInternal;
+
+  /*
+    CallBacks
+  */
+  const handleShare = (item: ArticleJSON) => {
+    // TODO - Implement the share function
+    console.log(item.url);
+    console.warn('To be developed');
+  };
+
+  /*
+    Renders
+  */
+
+  const renderHeader = () => {
+    return (
+      <RowBetween>
+        <HeaderButton icon="arrowLeft" onPress={goBack} />
+        <Row>
+          <HeaderButton
+            icon="share"
+            onPress={() => {
+              handleShare(news);
+            }}
+          />
+          <Separator width={18} />
+          <HeaderButton icon="paperPlane" onPress={() => {}} />
+        </Row>
+      </RowBetween>
+    );
+  };
+
+  const renderNewsHeaderSection = () => {
+    return (
+      <>
+        <Text typography="h8" color="gray">
+          📰 {news.source.name}
+        </Text>
+        <Text typography="h6" color="text" children={news.title} />
+      </>
+    );
+  };
+
+  const renderRedatorSection = () => {
+    const formatedDate = format(new Date(news.publishedAt), 'dd/MM/yyyy');
+    const hasAuthor = news.author === '' || !news.author ? false : true;
```

Comment:
You can make both this consts into 2 useMemos. that way, this info can be memoized to avoid unnecessary rerenders and recomputing


```suggestion
    const formatedDate = useMemo(()=>{
   return format(new Date(news.publishedAt), 'dd/MM/yyyy');
    },[news])
    
    const hasAuthor = useMemo(()=>{
    return  news.author === '' || !news.author ? false : true;
    },[news])
```